components/ai/code/ai-bbe/why-is-ballerina-the-way-you-should-write-ai-applications.md

```diff
@@ -1,61 +1,43 @@
-    audio:CreateTranscriptionResponse transcriptionRes = check openAIAudio->/audio/translations.post(translationsReq);
-    io:println("Audio text in English: ", transcriptionRes.text);
-
-    final chat:Client openAIChat = check new ({
-        auth: {
-            token: openAIToken
-        }
-    });
-
-    string query = string `Translate the following text from English to ${toLanguage} : ${transcriptionRes.text}`;
-
-    // Creates a request to translate the text from English to another language
-    chat:CreateChatCompletionRequest request = {
-        model: "gpt-4o",
-        messages: [
-            {
-                "role": "user",
-                "content": query
-            }
-        ],
-        temperature: 0.7,
-        max_tokens: 256,
-        top_p: 1,
-        frequency_penalty: 0,
-        presence_penalty: 0
-    };
-
-    // Translates the text from English to another language
-    chat:CreateChatCompletionResponse response = check openAIChat->/chat/completions.post(request);
-    string? translatedText = response.choices[0].message.content;
-
-    if translatedText is () {
-        return error("Failed to translate the given audio.");
+service /agent on hrAgent {
+    resource function post chat(@http:Payload agent:ChatReqMessage request) returns agent:ChatRespMessage|error {
+        string query = request.message;
+        embeddings:Deploymentid_embeddings_body embeddingsBody = {input: query};
+        embeddings:Inline_response_200 embeddingsResult = check azureOpenAIEmbeddings->/deployments/[EMBEDDING_DEPLOYMENT_ID]/embeddings.post("2023-03-15-preview", embeddingsBody);
+
+        float[] embeddingsFloat = decimalToFloatArray(embeddingsResult.data[0].embedding);
+        vector:QueryRequest queryRequest = {
+            topK: 4,
+            includeMetadata: true,
+            vector: embeddingsFloat
+        };
+        vector:QueryResponse response = check pinecodeVector->/query.post(queryRequest);
+        vector:QueryMatch[] matches = check response.matches.ensureType();
+
+        string context = check augment(matches);
+
+        string systemPrompt = string `You are an HR Policy Assistant that provides employees with accurate answers
+        based on company HR policies.Your responses must be clear and strictly based on the provided context.
+        ${context}`;
+
+        chat:CreateChatCompletionRequest chatRequest = {
+            messages: [
+                {role: "system", "content": systemPrompt},
+                {role: "user", "content": query}
+            ]
+        };
+
+        chat:CreateChatCompletionResponse chatResult =
+        check azureOpenAIChat->/deployments/[LLM_DEPLOYMENT_ID]/chat/completions.post("2023-12-01-preview", chatRequest);
+        ChatResponseChoice[] choices = check chatResult.choices.ensureType();
+        string chatResponse = check choices[0].message?.content.ensureType();
+
+        return {message: chatResponse};
```

Comment:
Can we think of bit simplied sample? this is bit too lenghthy and diagram is also not clear.

Reply:
We wanted a RAG sample since it's the 80% usecase.  One option is to keep the same scenario but divide it to blocks. Let me think of something